test/TestExtensions/API.Test/InternalAPITestHook.cs

```diff
@@ -180,14 +181,10 @@ public static void ProjectCacheEventApi_DetachHandler()
             vsSolutionManager.AfterNuGetCacheUpdated -= _cacheUpdateEventHandler.Invoke;
         }
 
-        public static IVsProjectJsonToPackageReferenceMigrateResult MigrateJsonProject(string projectName)
+        public static async Task<IVsProjectJsonToPackageReferenceMigrateResult> MigrateJsonProject(string projectName)
         {
             var migrator = ServiceLocator.GetInstance<IVsProjectJsonToPackageReferenceMigrator>();
-            return ThreadHelper.JoinableTaskFactory.Run(async () =>
-            {
-                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                return migrator.MigrateProjectJsonToPackageReference(projectName);
-            });
+            return await migrator.MigrateProjectJsonToPackageReferenceAsync(projectName) as IVsProjectJsonToPackageReferenceMigrateResult;
```

Comment:
If this method you're calling returned something that failed to cast, would you rather throw here or quietly return null? I suspect you'd rather throw. So why not use the `()` cast operator?